packages/dds/tree/docs/storage/treeStorage.md

```diff
@@ -0,0 +1,803 @@
+
+### Copy-On-Write History
+
+The need for "spine replacement" has a useful side effect: it makes it very simple to keep track of [revisions](#glossary) of the tree over time, i.e. the tree's [history](#glossary). Consider a tree made up of five chunks:
+
+```mermaid
+graph TD;
+    subgraph Revision 0:
+        A(Chunk A)
+        B(Chunk B)
+        C(Chunk C)
+        D(Chunk D)
+        E(Chunk E)
+        A-->B
+        A-->C
+        A-->E
+        C-->D
+    end
+```
+
+First, a client makes an edit to a node in Chunk B, so Chunk B and Chunk A (the spine) are replaced with Chunk B' and Chunk A'. This produces the next revision of the tree, revision 1:
+
+```mermaid
+graph TD;
+    subgraph Revision 1:
+        A(Chunk A')
+        B(Chunk B')
+        C(Chunk C)
+        D(Chunk D)
+        E(Chunk E)
+        A-->B
+        A-->C
+        A-->E
+        C-->D
+        style A fill:#844
+        style B fill:#844
+    end
+```
+
+Next, the client [edits](#glossary) Chunk D, resulting in the replacement of A', C, and D to form revision 2:
+
+```mermaid
+graph TD;
+    subgraph Revision 2:
+        A(Chunk A'')
+        B(Chunk B')
+        C(Chunk C')
+        D(Chunk D')
+        E(Chunk E)
+        A-->B
+        A-->C
+        A-->E
+        C-->D
+        style A fill:#448
+        style B fill:#844
+        style C fill:#448
+        style D fill:#448
+    end
+```
+
+Consider Chunk A and Chunk A', the predecessors to A''. They still exist in storage even though they are outdated and have been superceded by A'' in the latest revision of the tree (Revision 2). They are immutable and still retain the same contents that they did in prior revisions (their children are Chunk B/C/E and Chunk B'/C/E, respectively). Likewise, Chunk B, the predecessor to B', still exists (and the same for chunks C and D). In fact, _all_ chunks from all revisions remain accessible and untouched from their original state. This means that one can walk down from the root chunk of any revision and retrieve the entire contents of the tree as it had been at that revision. And conveniently, every revision produces exactly one new root chunk because a spine was cloned for each and the root chunk always belongs to any possible spine. This means that every revision is "captured" by a unique root chunk, and preserving the history of the tree can be accomplished by keeping references those root chunks. Storing the entire history of the tree is as simple as keeping a log of keys to those root chunks:
```

Comment:
`superceded` -> `superseded`
Supercede has occurred as a spelling variant of supersede since the 17th century, and it is common in current published writing. It continues, however, to be widely regarded as an error.

Reply:
Okay, thanks, good to know :) There is also the [spelling preferred by gardeners and landscapers](https://yewtreegc.co.uk/wp-content/uploads/2020/03/Evergreen-super-seed.jpg), although it is even more modern.

Reply:
I mean... we *are* trying to grow some big trees here.